deepali/losses/functional.py

```diff
@@ -638,6 +640,60 @@ def ssd_loss(
     return loss
 
 
+def sample_index_from_mask(mask: Tensor, num_samples: int):
```

Comment:
Could we possibly move `sample_index_from_mask` to `deepali.core.random` and name it `multinomial`? As it is a wrapper around `torch.multinomial` which only extends its functionality to a larger number of categories (for `replacement=True`). We can extend this later also to include the sampling without replacement using the Gumbel-max trick.

I think we can leave the function as is without moving and renaming it for this PR and do this in a separate PR.